packages/node_modules/@ciscospark/react-component-adaptive-card/src/index.js

```diff
@@ -17,11 +21,53 @@ class AdaptiveCard extends React.Component {
   componentDidMount() {
     if (Object.prototype.hasOwnProperty.call(this, 'nodeElement') && Object.prototype.hasOwnProperty.call(this.nodeElement, 'current')) {
       this.nodeElement.current.appendChild(
-        getAdaptiveCard(this.props.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
+        getAdaptiveCard(this.state.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
       );
     }
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.verb === API_ACTIVITY_VERB.SHARE) {
+      try {
+        const decryptedURLs = nextProps.items
+          .filter((file) => file.type === CARD_CONTAINS_IMAGE)
+          .map((file) => {
+            const thumbnail = file.mimeType === 'image/gif' ? nextProps.share.getIn(['files', file.url]) : nextProps.share.getIn(['files', file.image.url]);
+
+            if (thumbnail) {
+              const objectUrl = thumbnail.get('objectUrl');
+
+              return objectUrl;
+            }
+
+            return undefined;
+          });
+        const cards = JSON.parse(nextProps.cards[0]);
+        const undefinedUrls = decryptedURLs.filter((file) => file === undefined);
+
+        if (undefinedUrls.length === 0) {
+          const urlTypes = ['url', 'backgroundImage', 'iconUrl'];
+          const replacedImagesinCards = replaceIndexWithBlobURL(urlTypes, cards, decryptedURLs);
```

Comment:
Is there a reason why the `decryptedURLs` aren't calculated within `replaceIndexWithBlobURL`? I don't see anything on the calculation of the URL that's dependent on the component code.
It's a bit hard to follow the logic when half of it it's done in the component (the URL calculation) and the other half it's done in a helper function (the attribute replacement)

Reply:
There is no particular reason,`replaceIndexWithBlobURL` function is only for replacing the value so remaining code for URL calculation is in the component

Reply:
I think the URL calculation logic should be moved to `replaceIndexWithBlobURL`, after all the function name already describes that it will change index => URL.

Reply:
`replaceIndexWithBlobURL` is a recursive function.If we move URL calculation to  `replaceIndexWithBlobURL` then unnecessarily code will execute multiple times

Reply:
Im saying that in you only do the calculation once you get to the base condition, instead of pre-calculating everything, running multiple times through the card (you're traversing the cards once per filter operation)

Reply:
There is no base condition for URL calculation as we required updated `props` every time